src/legacy/core_plugins/expressions/public/np_ready/public/loader.ts

```diff
@@ -88,32 +90,40 @@ export class ExpressionLoader {
     return this.dataHandler.inspect();
   }
 
-  update(expression: string | ExpressionAST, params: IExpressionLoaderParams): Promise<RenderId> {
-    const promise = this.render$.pipe(first()).toPromise();
-
+  update(expression: string | ExpressionAST, params: IExpressionLoaderParams): void {
     if (expression !== null) {
-      this.execute(expression, params);
+      this.loadData(expression, params);
     } else {
       this.render(this.data);
     }
-    return promise;
   }
 
-  private execute = async (
+  private loadData = async (
     expression: string | ExpressionAST,
     params: IExpressionLoaderParams
   ): Promise<Data> => {
     if (this.dataHandler) {
       this.dataHandler.cancel();
     }
-    this.dataHandler = execute(expression, params);
-    const data = await this.dataHandler.getData();
-    this.dataSubject.next(data);
-    return data;
+    this.dataHandler = new ExpressionDataHandler(expression, params);
+    try {
+      const data = await this.dataHandler.getData();
+      if (data.type === 'error') {
+        if (data.error.name !== 'AbortError') {
+          this.dataSubject.error(new Error(data.error.message));
+          return;
+        }
+
+        return;
+      }
+      this.dataSubject.next(data);
+    } catch (e) {
+      this.dataSubject.error(new Error('Could not fetch data'));
```

Comment:
i think it would be nicer if we do not "eat" the error here, but pass it on.

Reply:
I wonder if we should keep `e` around in the new Error object, so it can be accessed and a deep-dive into the actual error could be determined?

Reply:
only thing i was suggesting is to `this.dataSubject.error(e)` instead of passing new Error `could not fetch data`

Reply:
I think we resolved this in the latest, errors are getting passed on